src/MQTTAsyncUtils.c

```diff
@@ -3086,7 +3122,7 @@ static MQTTPacket* MQTTAsync_cycle(int* sock, unsigned long timeout, int* rc)
 				}
 			}
 			else if (pack->header.bits.type == PUBREL)
-				*rc = MQTTProtocol_handlePubrels(pack, *sock);
+				*rc = MQTTProtocol_handlePubrels(pack, *sock, 1);
```

Comment:
To avoid a leak of the Topic Name, MQTTProtocol_handlePubrels must delete the topic name in addition to the pack upon its completion.